gst/nnstreamer/nnstreamer_plugin_api_impl.c

```diff
@@ -1428,3 +1428,91 @@ gst_tensor_meta_info_append_header (GstTensorMetaInfo * meta, GstMemory * mem)
   gst_memory_unmap (new_mem, &new_map);
   return new_mem;
 }
+
+#define NNS_TENSOR_EXTRA_MAGIC 0xf00dc0de
+
+/**
+ * @brief TBU
+*/
+void
+gst_tensors_extra_init (GstTensorExtraInfo * extra)
+{
+  guint i;
+
+  extra->magic = NNS_TENSOR_EXTRA_MAGIC;
+  extra->version = 0;
+  extra->num_extra_tensors = 0;
+  for (i = 0; i < NNS_TENSOR_SIZE_EXTRA_LIMIT; ++i) {
+    gst_tensor_info_init (&extra->infos[i]);
+  }
+}
+
+/**
+ * @brief TBU
+*/
+GstMemory *
+gst_tensors_get_nth_memory (GstBuffer * buffer, guint index)
+{
+  guint buffer_n_memory, i, offset = 0;
+  GstMemory *extra_tensors_memory, *res_mem;
+  GstMapInfo extra_tensors_map;
+  GstTensorExtraInfo *extra_info;
+
+  buffer_n_memory = gst_buffer_n_memory (buffer);
+
+  /* If buffer contains less than NNS_TENSOR_SIZE_LIMIT, it's trivial. */
+  if (buffer_n_memory < NNS_TENSOR_SIZE_LIMIT
+      || index < NNS_TENSOR_SIZE_LIMIT - 1) {
+    return gst_buffer_get_memory (buffer, index);
+  }
```

Comment:
As @jaeyun-jung said, if ```buffer_n_memory == 16``` and ```index == 15```, without the num_tensors, you don't know whether you may do the "quick access (traditional)" or "extended access".